Sources/Core/Cache.swift

```diff
@@ -0,0 +1,113 @@
+public typealias Size = Int
+
+public protocol Cacheable {
+    func cacheSize() ->  Size
+}
+
+public protocol CacheProtocol {
+    associatedtype Wrapped: Cacheable
+    init(maxSize: Size)
+    subscript(key: String) -> Wrapped? { get set }
+}
+
+public final class Cache<Wrapped: Cacheable>: CacheProtocol {
```

Comment:
I think we could do with just the concrete type. Since the protocol doesn't support serialization only a memory based implementation makes sense.